### DISCUSSION TITLE
P2P Swarm Problem

### DISCUSSION THREAD

**Discussion by rkdals03** [bitcraze/discussions#888] [Q&A]
Hi, I recently asked a question about the P2P swarm. There has been some progress thanks to this.

But if I run the code I modified, only one Drone flies, and the rest doesn't fly. I will upload the code I modified below and the demonstration video together. Please let me know what you think about this.

previous question : [https://github.com/orgs/bitcraze/discussions/842](https://github.com/orgs/bitcraze/discussions/842)
demo video : [https://youtu.be/OrlktIkUHjM?feature=shared](https://youtu.be/OrlktIkUHjM?feature=shared)
source code :  [https://github.com/DGIST-CPSec/ASDID/blob/main/Analyze_code/p2p_test.py](https://github.com/DGIST-CPSec/ASDID/blob/main/Analyze_code/p2p_test.py)

I look forward to hearing from you.
Many thanks. :)

**Comment by krichardsson** [comment]
Hi,
it is unfortunately impossible for us to debug that much code, sorry!

I recommend that you try to break the problem down in smaller bits and make sure each part works.
Another option is to start from something that works (I believe you started from an old demo?) and modify one small piece at a time and make sure that works before moving on to the next modification. 

Good luck!

(by the way, I updated your links, they did not work when clicked)

**Reply by rkdals03** [reply]
Thank you for comment !

But I don't know how to divide the code and run it.

Click the link now to open it.

I wonder if there is a tutorial related to flying using P2P. It's hard to fix it just by looking at the demo code because I'm not good enough. I'm sorry :(

**Reply by krichardsson** [reply]
I see. Have you managed to get the original demo running (without any modifications)? This would be a good base to have a solid ground to start from. 
If you have something that works you can make small changes and verify that they work. As soon as something works commit it to git to make sure you save the working state.
Also, if you commit your code to git often, you can easily check what has changes since it worked by using the [`git diff`](https://git-scm.com/docs/git-diff) command. `git diff` is also good if you want to ask questions here in discussions as it will enable you to show what has changed instead of providing the full file. 
If you could diff your file from our original file I could take a look (if it is not too many changes :-) )

**Reply by rkdals03** [reply]
Thank you for your advice. 
When I turned the demo code without modification, it didn't work because there was no charging pad.

I'll keep trying.
